src/Castle.Windsor/MicroKernel/Arguments.cs

```diff
@@ -29,25 +29,18 @@ public class Arguments : IDictionary
 		private bool isReadOnly;
 		private readonly IDictionary arguments;
 
-		public Arguments(IDictionary values, params IArgumentsComparer[] customComparers)
-			: this(customComparers)
+		public Arguments(IDictionary values)
+			: this()
 		{
 			foreach (DictionaryEntry entry in values)
 			{
 				Add(entry.Key, entry.Value);
 			}
 		}
 
-		public Arguments(params IArgumentsComparer[] customComparers)
+		public Arguments()
 		{
-			if (customComparers == null || customComparers.Length == 0)
-			{
-				arguments = new Dictionary<object, object>(new ArgumentsComparer());
-			}
-			else
-			{
-				arguments = new Dictionary<object, object>(new ArgumentsComparerExtended(customComparers));
-			}
+			arguments = new Dictionary<object, object>(new ArgumentsComparer());
```

Comment:
Where I went wrong before. I tried to get rid of the ArgumentsComparer by converting `Dictionary<object, object>` to `Dictionary<string, object>` so I could use `StringComparer` instead. ArgumentsComparer needs to hang around for a bit.